### ISSUE TITLE
Automatic Migration of labels when changing them in settings

### ISSUE THREAD

**Issue by Ierezell** [heartexlabs/label-studio#1620]
**Is your feature request related to a problem? Please describe.**
It's painful to have to delete all related annotations when changing a label. 

Indeed, when changing a label in the `settings` tab, an error raise and says : 
```
These labels still exist in annotations:
My Old Wonderful Label(12 annotations)
```
**Describe the solution you'd like**
Having an automatic migration that would replace all `My Old Wonderful Label` labels with `My New Wonderful Label`. 

**Describe alternatives you've considered**
Deleting all annotations and start labeling again...... 

**Additional context**
It's for text-based labeling so this means currently deleting annotations of all the spans, a migration would save so much time when changing settings...

**Comment by makseq** [comment]
Relative slack: https://label-studio.slack.com/archives/C01SKFX54QK/p1637079733054000

**Comment by ijmiller2** [comment]
Upvoting here as I experienced a bug (getting a validation error when trying to delete a label in the interface) related to this feature. See Slack post above from @makseq.

Thank you!
Ian

**Comment by priamaniv** [comment]
I experienced the same issue, upvoting and hoping for a fix soon!

**Comment by triklozoid** [comment]
@priamaniv we already have PR with the fix of this issue, you can check it, or wait for a little, until we release it. 
https://github.com/heartexlabs/label-studio/pull/1823

**Comment by ijmiller2** [comment]
@triklozoid , thanks for working on this issue. Just wanted to follow up to say that that while I am now able to save the Labeling Interface config when updating the label name, I still get a validation error complaining that the tasks with the old label still exists, even though I have manually filtered for tasks with this old label and deleted / updated them with the new label. 

Here's the error message I'm seeing, where I've replaced the actual label name with "previous_label_name":

```
Validation error 1526bff4-a899-4403-bdcc-1424936d3300
These labels still exist in annotations:
previous_label_name (1 annotations)
```

Just to be clear, when I go to the data manager and filter by "Annotation results" -> "contains" -> "previous_label_name". I get "Nothing found", as in no related tasks for the old label.

For reference, I'm using the enterprise version of the app. Happy to share the project ID or more info about the label privately if that would be helpful.

**Comment by triklozoid** [comment]
@ijmiller2 do you get this message after clicking on the "Save" button?  Now we have a workaround, which allows you to save config, but we are still working on fixing the reason of appearing this error message.

**Comment by ijmiller2** [comment]
@triklozoid, yes I see the error message after I click the "Save" button, but it persists after that. I just went back to the Labeling Interface / Editor and I'm still seeing that validation error there. I think the only way to get rid of it is to add the old / deleted label back.

**Comment by triklozoid** [comment]
@ijmiller2 can you please find me in our slack community(there is a link in README of this repo) and share the project id and label name?

**Comment by ijmiller2** [comment]
@triklozoid, no problem! Just sent you a message there

**Comment by makseq** [comment]
I've implemented label renaming in this PR as experimental feature 
https://github.com/heartexlabs/label-studio/pull/2055/files

You can switch to this branch and run LS as follow:
```
EXPERIMENTAL_FEATURES=1 label-studio
```

Please check this video tutorial: 
https://user-images.githubusercontent.com/501780/155040602-80e5158f-161b-4422-ae96-fa4c8c69e86d.mp4

**Comment by csaroff** [comment]
@makseq I tried out this feature, but unfortunately the project summary isn't correctly updated at the end.

At the end, I receive errors in the labeling interface config(/projects/<project_id>/settings/labeling) like this:

```python
Validation error f44f51b1-29d6-4ba2-9545-e60f8bf5d919
These labels still exist in annotations:
ABC (206 annotations)
Def Ghi (565 annotations)
```

After renaming "ABC" -> "abc" and "Def Ghi" -> "def_ghi".

I checked the labels themselves and I can't find any examples of "ABC" or "Def Ghi" so I think just the project summary is broken.

I don't know exactly what causes this broken state, but any calls to `/api/dm/actions?id=rename_labels` with sufficient labels times out after 30 seconds(I think the timeout is coming from our loadbalancer, but again not 100% sure).

Additionally, I ran the rename_labels api calls multiple times for the each label.

Is there a way to reset the project summary when it gets into this state?
